phone-book/src/edit-user.js

```diff
@@ -0,0 +1,195 @@
+
+
+class EditUser {
+  constructor(app) {
+    this.title = 'Add User'
+    this.app = app;
+    this.appContainer = app.appContainer;
+  }
+
+  initEvents(){
+    // this.createPlaceholders();
+    this.initSaveEvent();
+    if(this.app.state.userId){
+      this.initDeleteEvent();
+      this.renderUserData();
+    }
+  }
+
+  setCaretToEnd(node) {
+    let range = document.createRange();
+    let sel = window.getSelection();
+    range.setStart(node, 1);
+    range.collapse(true);
+    sel.removeAllRanges();
+    sel.addRange(range);
+  }
+  renderUserData(){
+    const contactObj = this.app.pages.contacts.pageObject;
+    const userId = this.app.state.userId;
+    this.getUser();
+  }
+
+  getUser(){
+    const userId = this.app.state.userId;
+    const serverAPI = this.app.serverAPI;
+    const loadingUser = serverAPI.getUser(userId);
+    loadingUser.then( response => {
+      return response.json();
+    })
+    .then(user => {
+      const fullNameField = this.appContainer.querySelector('span[data-name="fullName"]');
+      fullNameField.textContent = user.fullName;
+
+      const emailField = this.appContainer.querySelector('span[data-name="email"]');
+      emailField.textContent = user.email;
+
+      const phoneField = this.appContainer.querySelector('span[data-name="phone"]');
+      phoneField.textContent = user.phone;
+    })
+    .catch( err => {
+      console.error(err.message);
+    });
+
+  }
+
+  updateUser(data){
+    const serverAPI = this.app.serverAPI;
+    const updatingUser = serverAPI.updateUser(data);
+    updatingUser.then( response => {
+     this.app.pages.contacts.pageObject.users = [];
```

Comment:
it's breaking single responsibility principal an could causes unpredictable errors in future